### PR TITLE
Update finalhandler type definitions.

### PR DESCRIPTION
**Pull request by hbomark**
Used the repo of finalhandler as a resource, per link in checklist below.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

(updating existing definitions)
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/pillarjs/finalhandler/blob/v1.1.1/index.js#L84
- [x] Increase the version number in the header if appropriate. (set to v1.1.1)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. (tslint already exists)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1443,7 +1443,7 @@
 /types/fill-pdf/                                    @westy92
 /types/filter-console/                              @BendingBender
 /types/filter-invalid-dom-props/                    @icopp
-/types/finalhandler/                                @chrootsu
+/types/finalhandler/                                @pillarjs
```
Comment (uniqueiniquity) [inline .github/CODEOWNERS]: You don't need to edit this file directly; we'll update it in bulk based on your change to the header of `index.d.ts`.

File: types/finalhandler/index.d.ts [hunk 2]
```diff
@@ -1,20 +1,21 @@
-// Type definitions for finalhandler
+// Type definitions for finalhandler v1.1.1
```
Comment (uniqueiniquity) [inline types/finalhandler/index.d.ts]: This appears to be a malformed header. Could you remove the line in the `tslint.json` file in this folder that says `dt-header: false` and fix any errors that come up? You will want this first line to say `Type definitions for finalhandler 1.1`. You are welcome to remove any other such line from the `tslint.json` file, though that might be a lot of work depending on what issues are being ignored. Thanks!

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (hbomark) [reply .github/CODEOWNERS]: Removed this change.

File: types/finalhandler/index.d.ts (hunk 2 above)
Comment (hbomark) [reply types/finalhandler/index.d.ts]: I deleted all the `rules` from `tslint` and fixed all the linting problems, should be good now.

**Comment by hbomark** [comment]
Reran `npm lint finalhandler` and `npm test` before pushing up again.

Thank you so much @uniqueiniquity for your review!

**Review by uniqueiniquity — APPROVED** [review]

**Pull request merged**